test/functions/math.test.cpp

```diff
@@ -243,6 +243,30 @@ TEST_CASE("sqrt() of big integers", "[functions][math][sqrt][big]") {
             "098523490875239876209348752098273450239487562343");
 }
 
+TEST_CASE("is_probable_prime(10) for base cases", "[functions][math][is_probable_prime]") {
+    BigInt num = 1;
+    REQUIRE(num.is_probable_prime(10) == false);
+    num = 4;
+    REQUIRE(num.is_probable_prime(10) == false);
+    num = 2;
+    REQUIRE(num.is_probable_prime(10) == true);
+    num = 3;
+    REQUIRE(num.is_probable_prime(10) == true);
+}
+
+TEST_CASE("is_probable_prime(10/1) for big integers", "[functions][math][is_probable_prime][big]"){
+    BigInt num = 30121;
+    REQUIRE(num.is_probable_prime(10) == false); 
+    num = 564654598;
+    REQUIRE(num.is_probable_prime(10) == false);
+    num = 5915587277;
+    REQUIRE(num.is_probable_prime(10) == true);
+    num = "4702877858580764415667235078667510929270158248348";
+    REQUIRE(num.is_probable_prime(1) == false);
+    num = "378348910233465647859184421334615532543749747185321634086219";
+    REQUIRE(num.is_probable_prime(1) == true);
```

Comment:
It would be good to have checks for even larger numbers, and with more variation in the value of certainty.

Reply:
Using bigger numbers and raising certainty will raise the time dramatically but I will implement both.

Reply:
> Using bigger numbers and raising certainty will raise the time dramatically

Yes, that's why we're testing it with bigger values - to check that it shouldn't take too long to work out the result for much larger numbers.

Reply:
Understood, because of the way the algorithm works with certainty. It's going to take a long time unless we optimize the operations for BigInt. I agree with you though, I believed that it shouldn't take too long to work out result for much larger numbers. How would you like to tackle the problem?